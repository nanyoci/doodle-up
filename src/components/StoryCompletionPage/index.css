.story-completion-cover {
	display: block;
	margin: 0 auto;
}