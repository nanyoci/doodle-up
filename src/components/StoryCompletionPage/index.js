import React from 'react';
import { Link } from 'react-router-dom';
import { Button } from 'semantic-ui-react';

import Page from '../Page';
import './index.css';

export default function StoryCompletionPage({story}) {
	return (
		<Page>
			<h1>You've reached the end of {story.story_title}</h1>
			<img className="story-completion-cover" src={story.cover_image} />
		</Page>
	)
}
