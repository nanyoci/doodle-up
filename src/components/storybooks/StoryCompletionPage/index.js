import React from 'react';

import Page from '../../common/Page';
import './index.css';

export default function StoryCompletionPage({ story, isReadOnly, onReset }) {
	return (
		<Page>
			<h1>You've reached the end of {story.story_title}</h1>
			<img className="story-completion-cover" src={story.cover_image} alt={story.story_title} />
			{
				!isReadOnly &&
				<div className="text-center mt-4">
					<div className="menu-buttons mb-2">
						<button className="btn btn-primary" onClick={onReset}>Play again?</button>
					</div>
					<p>Take note that this will delete your progress and previous drawings.</p>
				</div>
			}
		</Page>
	)
}
