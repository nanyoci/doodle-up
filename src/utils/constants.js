export const API_URL = 'https://doodle-up-backend.herokuapp.com';