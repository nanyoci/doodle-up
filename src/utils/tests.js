import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';

import rootReducer from '../redux/rootReducer';

export const renderWithRedux = (component, initialState = {}) => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )

    const renderedComponent = <Provider store={store}>{component}</Provider>;

    return {
        ...render(renderedComponent),
        store
    };
}

export const renderWithRouter = (component, route = '/') => {
    const history = createMemoryHistory({ initialEntries: [route] });
    const renderedComponent = <Router history={history}>{component}</Router>;

    return {
        ...render(renderedComponent),
        history
    };
}

export const renderWithReduxRouter = (component, initialState = {}, route = '/') => {
    const store = createStore(
        rootReducer,
        initialState,
        applyMiddleware(thunk),
    )
    const history = createMemoryHistory({ initialEntries: [route] });

    const renderedComponent = (
        <Provider store={store}>
            <Router history={history}>{component}</Router>
        </Provider>
    );

    return {
        ...render(renderedComponent),
        history,
        store
    };
}