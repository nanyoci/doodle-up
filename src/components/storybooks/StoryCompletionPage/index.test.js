import React from 'react'
import { cleanup, act } from '@testing-library/react'

import cover from './sample/cover.png';
import StoryCompletionPage from './index.js';
import { renderWithReduxRouter } from './../../../utils/tests';

jest.mock('axios');

beforeEach(() => {
	window.HTMLMediaElement.prototype.load = () => { /* do nothing */ };
	window.HTMLMediaElement.prototype.play = () => { /* do nothing */ };
	window.HTMLMediaElement.prototype.pause = () => { /* do nothing */ };
	window.HTMLMediaElement.prototype.addTextTrack = () => { /* do nothing */ };
});

afterEach(() => {
	cleanup();
	jest.clearAllMocks();
});

const story = {
	story_title: "Three Little Pigs",
	cover_image: cover,
}

it('should take a snapshot', async () => {
	let asFragment = {};
	await act(async () => {
		({ asFragment } = renderWithReduxRouter(
			<StoryCompletionPage
				story={story}
				isReadOnly={false}
				onReset={() => { }}
			/>
		))
	});
	expect(asFragment()).toMatchSnapshot();
})
