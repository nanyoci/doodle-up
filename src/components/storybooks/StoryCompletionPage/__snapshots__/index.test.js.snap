// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="main-content"
  >
    <audio />
    <div
      class="wrapper"
    >
      <a
        href="/"
      >
        <img
          alt="DoodleUp"
          class="logo-corner"
          src="doodle-up.svg"
        />
      </a>
      <h1>
        You've reached the end of Three Little Pigs
      </h1>
      <img
        alt="Three Little Pigs"
        class="story-completion-cover"
        src="cover.png"
      />
      <div
        class="text-center mt-4"
      >
        <div
          class="menu-buttons mb-2"
        >
          <button
            class="btn btn-primary"
          >
            Play again?
          </button>
        </div>
        <p>
          Take note that this will delete your progress and previous drawings.
        </p>
      </div>
    </div>
    <div
      class="control-buttons"
    >
      <button
        class="ui yellow massive circular icon button"
        id="signOutBtn"
      >
        <i
          aria-hidden="true"
          class="sign-out icon"
        />
      </button>
      <button
        class="ui yellow massive circular icon button"
        id="musicControlBtn"
      >
        <i
          aria-hidden="true"
          class="mute icon"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
